### PR TITLE
Update f/e dependencies : part 1

### LINKED ISSUE #715: Package update
Time to bump dependencies - though best to wait after merging the feature branch

# Major
| Package | Current | Wanted | Latest | Package Type | URL | CHANGELOG |
|---|---|---|---|---|---|---|
| @awesome-cordova-plugins/core | 5.44.0 | 5.46.0 | 6.3.0 | @openmsupply-client/system | devDependencies | https://github.com/danielsogl/awesome-cordova-plugins#readme |  |
| @awesome-cordova-plugins/printer | 5.44.0 | 5.46.0 | 6.3.0 | @openmsupply-client/system | devDependencies | https://github.com/danielsogl/awesome-cordova-plugins#readme |  |
| @capacitor-community/barcode-scanner | 2.1.1 | 2.1.1 | 3.0.3 | openmsupply-client | devDependencies | https://github.com/capacitor-community/barcode-scanner#readme |  |
| @capacitor/android | 3.6.0 | 3.9.0 | 4.7.0 | @openmsupply-client/android | devDependencies | https://capacitorjs.com |  |
| @capacitor/cli | 3.6.0 | 3.9.0 | 4.7.0 | @openmsupply-client/android | devDependencies | https://capacitorjs.com |  |
| @capacitor/core | 3.6.0 | 3.9.0 | 4.7.0 | @openmsupply-client/android | devDependencies | https://capacitorjs.com |  |
| @capacitor/core | 3.6.0 | 3.9.0 | 4.7.0 | @openmsupply-client/common | dependencies | https://capacitorjs.com |  |
| @capacitor/keyboard | 1.2.3 | 1.2.3 | 4.1.1 | openmsupply-client | devDependencies | https://github.com/ionic-team/capacitor-plugins#readme |  |
| @graphql-codegen/cli | 2.7.0 | 2.16.5 | 3.2.2 | openmsupply-client | devDependencies | https://github.com/dotansimha/graphql-code-generator#readme |  |
| @graphql-codegen/typescript | 2.6.0 | 2.8.8 | 3.0.2 | openmsupply-client | devDependencies | https://github.com/dotansimha/graphql-code-generator#readme |  |
| @graphql-codegen/typescript-operations | 2.4.3 | 2.5.13 | 3.0.2 | openmsupply-client | devDependencies | https://github.com/dotansimha/graphql-code-generator#readme |  |
| @testing-library/dom | 8.14.0 | 8.20.0 | 9.0.1 | openmsupply-client | devDependencies | https://github.com/testing-library/dom-testing-library#readme |  |
| @testing-library/react | 13.3.0 | 13.4.0 | 14.0.0 | openmsupply-client | devDependencies | https://github.com/testing-library/react-testing-library#readme |  |
| @types/dompurify | 2.3.3 | 2.4.0 | 3.0.0 | @openmsupply-client/common | devDependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/dompurify |  |
| @types/jest | 28.1.4 | 28.1.4 | 29.4.1 | openmsupply-client | devDependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/jest |  |
| @types/uuid | 8.3.4 | 8.3.4 | 9.0.1 | @openmsupply-client/common | devDependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/uuid |  |
| babel-jest | 28.1.2 | 28.1.3 | 29.5.0 | openmsupply-client | devDependencies | https://github.com/facebook/jest#readme |  |
| babel-loader | 8.2.5 | 8.3.0 | 9.1.2 | openmsupply-client | devDependencies | https://github.com/babel/babel-loader |  |
| cordova-android | 10.1.2 | 10.1.2 | 11.0.0 | @openmsupply-client/android | devDependencies | https://github.com/apache/cordova-android#readme |  |
| dompurify | 2.3.8 | 2.4.5 | 3.0.1 | @openmsupply-client/common | dependencies | https://github.com/cure53/DOMPurify |  |
| electron | 19.0.8 | 19.0.8 | 23.1.3 | @openmsupply-client/electron | devDependencies | https://github.com/electron/electron#readme |  |
| eslint-plugin-storybook | 0.5.13 | 0.5.13 | 0.6.11 | openmsupply-client | devDependencies | https://github.com/storybookjs/eslint-plugin-storybook#readme |  |
| fork-ts-checker-webpack-plugin | 7.2.11 | 7.2.11 | 8.0.0 | @openmsupply-client/electron | devDependencies | https://github.com/TypeStrong/fork-ts-checker-webpack-plugin#readme |  |
| graphql-request | 4.3.0 | 4.3.0 | 5.2.0 | @openmsupply-client/common | dependencies | https://github.com/jasonkuhrt/graphql-request |  |
| i18n-unused | 0.8.0 | 0.8.0 | 0.13.0 | openmsupply-client | devDependencies | https://github.com/mxmvshnvsk/i18n-unused#readme |  |
| i18next | 21.8.13 | 21.10.0 | 22.4.11 | @openmsupply-client/common | dependencies | https://www.i18next.com |  |
| i18next-browser-languagedetector | 6.1.4 | 6.1.8 | 7.0.1 | @openmsupply-client/common | dependencies | https://github.com/i18next/i18next-browser-languageDetector |  |
| i18next-chained-backend | 3.0.2 | 3.1.0 | 4.2.0 | @openmsupply-client/common | dependencies | https://github.com/i18next/i18next-chained-backend |  |
| i18next-http-backend | 1.4.1 | 1.4.5 | 2.1.1 | @openmsupply-client/common | dependencies | https://github.com/i18next/i18next-http-backend |  |
| i18next-localstorage-backend | 3.1.3 | 3.1.3 | 4.1.0 | @openmsupply-client/common | dependencies | https://github.com/i18next/i18next-localStorage-backend |  |
| jest | 28.1.2 | 28.1.3 | 29.5.0 | openmsupply-client | devDependencies | https://jestjs.io/ |  |
| jest-environment-jsdom | 28.1.2 | 28.1.3 | 29.5.0 | openmsupply-client | devDependencies | https://github.com/facebook/jest#readme |  |
| lerna | 5.1.8 | 5.6.2 | 6.5.1 | openmsupply-client | dependencies | https://lerna.js.org |  |
| msw | 0.43.1 | 0.43.1 | 1.1.0 | openmsupply-client | dependencies | https://mswjs.io |  |
| notistack | 2.0.5 | 2.0.8 | 3.0.1 | @openmsupply-client/common | dependencies | https://www.notistack.com |  |
| react-i18next | 11.18.0 | 11.18.6 | 12.2.0 | @openmsupply-client/common | dependencies | https://github.com/i18next/react-i18next |  |
| react-viewport-list | 6.3.0 | 6.3.0 | 7.0.0 | @openmsupply-client/common | dependencies | https://codesandbox.io/s/react-viewport-list-xw2rt |  |
| serve | 13.0.4 | 13.0.4 | 14.2.0 | @openmsupply-client/host | devDependencies | https://github.com/vercel/serve#readme |  |
| ts-jest | 28.0.5 | 28.0.8 | 29.0.5 | openmsupply-client | devDependencies | https://kulshekhar.github.io/ts-jest |  |
| tsconfig-paths-webpack-plugin | 3.5.2 | 3.5.2 | 4.0.1 | openmsupply-client | devDependencies | https://github.com/dividab/tsconfig-paths-webpack-plugin |  |
| uuid | 8.3.2 | 8.3.2 | 9.0.0 | @openmsupply-client/common | dependencies | https://github.com/uuidjs/uuid#readme |  |
| webpack-cli | 4.10.0 | 4.10.0 | 5.0.1 | @openmsupply-client/host | devDependencies | https://github.com/webpack/webpack-cli/tree/master/packages/webpack-cli |  |
| zustand | 3.7.2 | 3.7.2 | 4.3.6 | @openmsupply-client/common | dependencies | https://github.com/pmndrs/zustand |  |

# Minor
| Package | Current | Wanted | Latest | Package Type | URL | CHANGELOG |
|---|---|---|---|---|---|---|
| @babel/core | 7.18.6 | 7.21.0 | 7.21.0 | openmsupply-client | devDependencies | https://babel.dev/docs/en/next/babel-core |  |
| @babel/preset-env | 7.18.6 | 7.20.2 | 7.20.2 | openmsupply-client | devDependencies | https://babel.dev/docs/en/next/babel-preset-env |  |
| @bugsnag/js | 7.17.0 | 7.20.0 | 7.20.0 | openmsupply-client | dependencies | https://www.bugsnag.com/ |  |
| @bugsnag/js | 7.17.0 | 7.20.0 | 7.20.0 | @openmsupply-client/common | dependencies | https://www.bugsnag.com/ |  |
| @emotion/react | 11.9.3 | 11.10.6 | 11.10.6 | @openmsupply-client/common | dependencies | https://github.com/emotion-js/emotion/tree/main#readme |  |
| @emotion/styled | 11.9.3 | 11.10.6 | 11.10.6 | @openmsupply-client/common | dependencies | https://github.com/emotion-js/emotion/tree/main#readme |  |
| @graphql-codegen/near-operation-file-preset | 2.2.15 | 2.5.0 | 2.5.0 | openmsupply-client | devDependencies | https://github.com/dotansimha/graphql-code-generator-community#readme |  |
| @graphql-codegen/typescript-graphql-request | 4.4.11 | 4.5.8 | 4.5.8 | openmsupply-client | devDependencies | https://github.com/dotansimha/graphql-code-generator#readme |  |
| @graphql-codegen/typescript-msw | 1.0.13 | 1.1.6 | 1.1.6 | openmsupply-client | devDependencies | https://github.com/dotansimha/graphql-code-generator#readme |  |
| @mui/material | 5.8.7 | 5.11.12 | 5.11.12 | @openmsupply-client/common | dependencies | https://mui.com/material-ui/getting-started/overview/ |  |
| @pmmmwh/react-refresh-webpack-plugin | 0.5.7 | 0.5.10 | 0.5.10 | @openmsupply-client/host | dependencies | https://github.com/pmmmwh/react-refresh-webpack-plugin#readme |  |
| @swc/cli | 0.1.57 | 0.1.62 | 0.1.62 | openmsupply-client | devDependencies | https://github.com/swc-project/cli#readme |  |
| @swc/core | 1.2.211 | 1.3.40 | 1.3.40 | openmsupply-client | devDependencies | https://swc.rs |  |
| @swc/jest | 0.2.21 | 0.2.24 | 0.2.24 | openmsupply-client | devDependencies | https://github.com/swc-project/jest |  |
| @testing-library/user-event | 14.2.1 | 14.4.3 | 14.4.3 | openmsupply-client | devDependencies | https://github.com/testing-library/user-event#readme |  |
| @types/node | 18.0.3 | 18.0.3 | 18.15.2 | openmsupply-client | devDependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node |  |
| @typescript-eslint/eslint-plugin | 5.30.5 | 5.55.0 | 5.55.0 | openmsupply-client | devDependencies | https://github.com/typescript-eslint/typescript-eslint#readme |  |
| @typescript-eslint/eslint-plugin | 5.30.5 | 5.55.0 | 5.55.0 | @openmsupply-client/electron | devDependencies | https://github.com/typescript-eslint/typescript-eslint#readme |  |
| @typescript-eslint/parser | 5.30.5 | 5.55.0 | 5.55.0 | openmsupply-client | devDependencies | https://github.com/typescript-eslint/typescript-eslint#readme |  |
| @typescript-eslint/parser | 5.30.5 | 5.55.0 | 5.55.0 | @openmsupply-client/electron | devDependencies | https://github.com/typescript-eslint/typescript-eslint#readme |  |
| date-fns | 2.28.0 | 2.29.3 | 2.29.3 | @openmsupply-client/common | dependencies | https://github.com/date-fns/date-fns#readme |  |
| eslint | 8.19.0 | 8.36.0 | 8.36.0 | openmsupply-client | devDependencies | https://eslint.org |  |
| eslint | 8.19.0 | 8.36.0 | 8.36.0 | @openmsupply-client/electron | devDependencies | https://eslint.org |  |
| eslint-config-prettier | 8.5.0 | 8.7.0 | 8.7.0 | @openmsupply-client/common | devDependencies | https://github.com/prettier/eslint-config-prettier#readme |  |
| eslint-config-prettier | 8.5.0 | 8.7.0 | 8.7.0 | @openmsupply-client/config | devDependencies | https://github.com/prettier/eslint-config-prettier#readme |  |
| eslint-config-prettier | 8.5.0 | 8.7.0 | 8.7.0 | @openmsupply-client/dashboard | devDependencies | https://github.com/prettier/eslint-config-prettier#readme |  |
| eslint-config-prettier | 8.5.0 | 8.7.0 | 8.7.0 | @openmsupply-client/host | devDependencies | https://github.com/prettier/eslint-config-prettier#readme |  |
| eslint-config-prettier | 8.5.0 | 8.7.0 | 8.7.0 | @openmsupply-client/inventory | devDependencies | https://github.com/prettier/eslint-config-prettier#readme |  |
| eslint-config-prettier | 8.5.0 | 8.7.0 | 8.7.0 | @openmsupply-client/invoices | devDependencies | https://github.com/prettier/eslint-config-prettier#readme |  |
| eslint-config-prettier | 8.5.0 | 8.7.0 | 8.7.0 | @openmsupply-client/requisitions | devDependencies | https://github.com/prettier/eslint-config-prettier#readme |  |
| eslint-config-prettier | 8.5.0 | 8.7.0 | 8.7.0 | @openmsupply-client/system | devDependencies | https://github.com/prettier/eslint-config-prettier#readme |  |
| eslint-plugin-import | 2.26.0 | 2.27.5 | 2.27.5 | @openmsupply-client/electron | devDependencies | https://github.com/import-js/eslint-plugin-import |  |
| eslint-plugin-react | 7.30.1 | 7.32.2 | 7.32.2 | openmsupply-client | devDependencies | https://github.com/jsx-eslint/eslint-plugin-react |  |
| graphql | 16.5.0 | 16.6.0 | 16.6.0 | openmsupply-client | dependencies | https://github.com/graphql/graphql-js |  |
| lint-staged | 13.0.3 | 13.0.3 | 13.2.0 | @openmsupply-client/common | devDependencies | https://github.com/okonet/lint-staged#readme |  |
| lint-staged | 13.0.3 | 13.0.3 | 13.2.0 | @openmsupply-client/config | devDependencies | https://github.com/okonet/lint-staged#readme |  |
| lint-staged | 13.0.3 | 13.0.3 | 13.2.0 | @openmsupply-client/dashboard | devDependencies | https://github.com/okonet/lint-staged#readme |  |
| lint-staged | 13.0.3 | 13.0.3 | 13.2.0 | @openmsupply-client/host | devDependencies | https://github.com/okonet/lint-staged#readme |  |
| lint-staged | 13.0.3 | 13.0.3 | 13.2.0 | @openmsupply-client/inventory | devDependencies | https://github.com/okonet/lint-staged#readme |  |
| lint-staged | 13.0.3 | 13.0.3 | 13.2.0 | @openmsupply-client/invoices | devDependencies | https://github.com/okonet/lint-staged#readme |  |
| lint-staged | 13.0.3 | 13.0.3 | 13.2.0 | @openmsupply-client/requisitions | devDependencies | https://github.com/okonet/lint-staged#readme |  |
| lint-staged | 13.0.3 | 13.0.3 | 13.2.0 | @openmsupply-client/system | devDependencies | https://github.com/okonet/lint-staged#readme |  |
| msw-storybook-addon | 1.6.3 | 1.8.0 | 1.8.0 | openmsupply-client | devDependencies | https://msw-sb.vercel.app/ |  |
| papaparse | 5.3.2 | 5.4.0 | 5.4.0 | openmsupply-client | dependencies | http://papaparse.com |  |
| prettier | 2.7.1 | 2.8.4 | 2.8.4 | @openmsupply-client/common | devDependencies | https://prettier.io |  |
| prettier | 2.7.1 | 2.8.4 | 2.8.4 | @openmsupply-client/config | devDependencies | https://prettier.io |  |
| prettier | 2.7.1 | 2.8.4 | 2.8.4 | @openmsupply-client/dashboard | devDependencies | https://prettier.io |  |
| prettier | 2.7.1 | 2.8.4 | 2.8.4 | @openmsupply-client/host | devDependencies | https://prettier.io |  |
| prettier | 2.7.1 | 2.8.4 | 2.8.4 | @openmsupply-client/inventory | devDependencies | https://prettier.io |  |
| prettier | 2.7.1 | 2.8.4 | 2.8.4 | @openmsupply-client/invoices | devDependencies | https://prettier.io |  |
| prettier | 2.7.1 | 2.8.4 | 2.8.4 | @openmsupply-client/requisitions | devDependencies | https://prettier.io |  |
| prettier | 2.7.1 | 2.8.4 | 2.8.4 | @openmsupply-client/system | devDependencies | https://prettier.io |  |
| react-router-dom | 6.3.0 | 6.9.0 | 6.9.0 | @openmsupply-client/common | dependencies | https://github.com/remix-run/react-router#readme |  |
| react-router-dom | 6.3.0 | 6.9.0 | 6.9.0 | @openmsupply-client/host | dependencies | https://github.com/remix-run/react-router#readme |  |
| recharts | 2.1.12 | 2.4.3 | 2.4.3 | @openmsupply-client/common | dependencies | https://github.com/recharts/recharts |  |
| recharts | 2.1.12 | 2.4.3 | 2.4.3 | @openmsupply-client/dashboard | dependencies | https://github.com/recharts/recharts |  |
| ts-loader | 9.3.1 | 9.4.2 | 9.4.2 | @openmsupply-client/electron | devDependencies | https://github.com/TypeStrong/ts-loader |  |
| typescript | 4.7.4 | 4.9.5 | 4.9.5 | @openmsupply-client/common | devDependencies | https://www.typescriptlang.org/ |  |
| typescript | 4.7.4 | 4.9.5 | 4.9.5 | @openmsupply-client/config | devDependencies | https://www.typescriptlang.org/ |  |
| typescript | 4.7.4 | 4.9.5 | 4.9.5 | @openmsupply-client/dashboard | devDependencies | https://www.typescriptlang.org/ |  |
| typescript | 4.7.4 | 4.7.4 | 4.9.5 | @openmsupply-client/electron | devDependencies | https://www.typescriptlang.org/ |  |
| typescript | 4.7.4 | 4.9.5 | 4.9.5 | @openmsupply-client/host | devDependencies | https://www.typescriptlang.org/ |  |
| typescript | 4.7.4 | 4.9.5 | 4.9.5 | @openmsupply-client/inventory | devDependencies | https://www.typescriptlang.org/ |  |
| typescript | 4.7.4 | 4.9.5 | 4.9.5 | @openmsupply-client/invoices | devDependencies | https://www.typescriptlang.org/ |  |
| typescript | 4.7.4 | 4.9.5 | 4.9.5 | @openmsupply-client/requisitions | devDependencies | https://www.typescriptlang.org/ |  |
| typescript | 4.7.4 | 4.9.5 | 4.9.5 | @openmsupply-client/system | devDependencies | https://www.typescriptlang.org/ |  |
| webpack | 5.73.0 | 5.76.1 | 5.76.1 | openmsupply-client | devDependencies | https://github.com/webpack/webpack |  |
| webpack | 5.73.0 | 5.76.1 | 5.76.1 | @openmsupply-client/host | devDependencies | https://github.com/webpack/webpack |  |
| webpack-bundle-analyzer | 4.5.0 | 4.8.0 | 4.8.0 | @openmsupply-client/host | dependencies | https://github.com/webpack-contrib/webpack-bundle-analyzer |  |
| webpack-dev-server | 4.9.3 | 4.11.1 | 4.11.1 | @openmsupply-client/host | devDependencies | https://github.com/webpack/webpack-dev-server#readme |  |

# Patch
| Package | Current | Wanted | Latest | Package Type | URL | CHANGELOG |
|---|---|---|---|---|---|---|
| @electron-forge/cli | 6.0.0-beta.64 | 6.0.5 | 6.0.5 | @openmsupply-client/electron | devDependencies | https://github.com/electron/forge#readme |  |
| @electron-forge/maker-deb | 6.0.0-beta.64 | 6.0.5 | 6.0.5 | @openmsupply-client/electron | devDependencies | https://github.com/electron/forge#readme |  |
| @electron-forge/maker-rpm | 6.0.0-beta.64 | 6.0.5 | 6.0.5 | @openmsupply-client/electron | devDependencies | https://github.com/electron/forge#readme |  |
| @electron-forge/maker-squirrel | 6.0.0-beta.64 | 6.0.5 | 6.0.5 | @openmsupply-client/electron | devDependencies | https://github.com/electron/forge#readme |  |
| @electron-forge/maker-zip | 6.0.0-beta.64 | 6.0.5 | 6.0.5 | @openmsupply-client/electron | devDependencies | https://github.com/electron/forge#readme |  |
| @electron-forge/plugin-webpack | 6.0.0-beta.63 | 6.0.0-beta.63 | 6.0.5 | @openmsupply-client/electron | devDependencies | https://github.com/electron/forge#readme |  |
| @fontsource/inter | 4.5.11 | 4.5.15 | 4.5.15 | @openmsupply-client/host | dependencies | https://github.com/fontsource/fontsource/tree/main/fonts/google/inter#readme |  |
| @storybook/addon-actions | 6.5.9 | 6.5.16 | 6.5.16 | openmsupply-client | devDependencies | https://github.com/storybookjs/storybook/tree/main/addons/actions |  |
| @storybook/addon-essentials | 6.5.9 | 6.5.16 | 6.5.16 | openmsupply-client | devDependencies | https://github.com/storybookjs/storybook/tree/main/addons/essentials |  |
| @storybook/addon-interactions | 6.5.9 | 6.5.16 | 6.5.16 | openmsupply-client | devDependencies | https://github.com/storybookjs/storybook/tree/main/addons/interactions |  |
| @storybook/addon-links | 6.5.9 | 6.5.16 | 6.5.16 | openmsupply-client | devDependencies | https://github.com/storybookjs/storybook/tree/main/addons/links |  |
| @storybook/addons | 6.5.9 | 6.5.16 | 6.5.16 | openmsupply-client | devDependencies | https://github.com/storybookjs/storybook/tree/main/lib/addons |  |
| @storybook/builder-webpack5 | 6.5.9 | 6.5.16 | 6.5.16 | openmsupply-client | devDependencies | https://github.com/storybookjs/storybook/tree/main/lib/core |  |
| @storybook/manager-webpack5 | 6.5.9 | 6.5.16 | 6.5.16 | openmsupply-client | devDependencies | https://github.com/storybookjs/storybook/tree/main/lib/core |  |
| @storybook/react | 6.5.9 | 6.5.16 | 6.5.16 | openmsupply-client | devDependencies | https://github.com/storybookjs/storybook/tree/main/app/react |  |
| @storybook/theming | 6.5.9 | 6.5.16 | 6.5.16 | openmsupply-client | devDependencies | https://github.com/storybookjs/storybook/tree/main/lib/theming |  |
| @testing-library/jest-dom | 5.16.4 | 5.16.5 | 5.16.5 | openmsupply-client | devDependencies | https://github.com/testing-library/jest-dom#readme |  |
| @types/js-cookie | 3.0.2 | 3.0.3 | 3.0.3 | @openmsupply-client/common | devDependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/js-cookie |  |
| @types/lodash | 4.14.182 | 4.14.191 | 4.14.191 | openmsupply-client | devDependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/lodash |  |
| @types/papaparse | 5.3.2 | 5.3.7 | 5.3.7 | openmsupply-client | devDependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/papaparse |  |
| @types/react | 18.0.15 | 18.0.28 | 18.0.28 | openmsupply-client | devDependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react |  |
| @types/react-dom | 18.0.6 | 18.0.11 | 18.0.11 | openmsupply-client | dependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-dom |  |
| @types/recharts | 1.8.23 | 1.8.24 | 1.8.24 | openmsupply-client | devDependencies | https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/recharts |  |
| @vercel/webpack-asset-relocator-loader | 1.7.2 | 1.7.2 | 1.7.3 | @openmsupply-client/electron | devDependencies | https://github.com/vercel/webpack-asset-relocator-loader#readme |  |
| css-loader | 6.7.1 | 6.7.3 | 6.7.3 | @openmsupply-client/electron | devDependencies | https://github.com/webpack-contrib/css-loader |  |
| eslint-plugin-jest-dom | 4.0.2 | 4.0.3 | 4.0.3 | openmsupply-client | devDependencies | https://github.com/testing-library/eslint-plugin-jest-dom#readme |  |
| husky | 8.0.1 | 8.0.3 | 8.0.3 | @openmsupply-client/common | devDependencies | https://typicode.github.io/husky |  |
| husky | 8.0.1 | 8.0.3 | 8.0.3 | @openmsupply-client/config | devDependencies | https://typicode.github.io/husky |  |
| husky | 8.0.1 | 8.0.3 | 8.0.3 | @openmsupply-client/dashboard | devDependencies | https://typicode.github.io/husky |  |
| husky | 8.0.1 | 8.0.3 | 8.0.3 | @openmsupply-client/host | devDependencies | https://typicode.github.io/husky |  |
| husky | 8.0.1 | 8.0.3 | 8.0.3 | @openmsupply-client/inventory | devDependencies | https://typicode.github.io/husky |  |
| husky | 8.0.1 | 8.0.3 | 8.0.3 | @openmsupply-client/invoices | devDependencies | https://typicode.github.io/husky |  |
| husky | 8.0.1 | 8.0.3 | 8.0.3 | @openmsupply-client/requisitions | devDependencies | https://typicode.github.io/husky |  |
| husky | 8.0.1 | 8.0.3 | 8.0.3 | @openmsupply-client/system | devDependencies | https://typicode.github.io/husky |  |
| react-currency-input-field | 3.6.4 | 3.6.10 | 3.6.10 | @openmsupply-client/common | dependencies | https://github.com/cchanxzy/react-currency-input-field |  |
| react-query | 3.39.1 | 3.39.3 | 3.39.3 | @openmsupply-client/common | dependencies | https://github.com/tannerlinsley/react-query#readme |  |
| storybook-addon-swc | 1.1.7 | 1.1.9 | 1.1.9 | openmsupply-client | devDependencies | https://github.com/Karibash/storybook-addon-swc |  |
| style-loader | 3.3.1 | 3.3.2 | 3.3.2 | @openmsupply-client/electron | devDependencies | https://github.com/webpack-contrib/style-loader |  |
| stylis | 4.1.1 | 4.1.3 | 4.1.3 | @openmsupply-client/common | dependencies | https://github.com/thysultan/stylis.js |  |

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #715 (partly)

# 👩🏻‍💻 What does this PR do? 
 
Upgrades patch and minor versions, and some of the major versions. 
Reached the point of upgrading zustand and thought that I'd remove  the deprecated methods too, which will be a little too much to add in to this PR.

Key changes:
- i18next has been bumped which has resulted in some typescript changes. Most notably, I've dropped the overloaded default value version of the `useTranslation` method and changed to use `defaultValue` as a property on the options.
- the react-viewport-list has been bumped by a major version. only a few minor breaking changes it is only implemented in DataTable so it is quick enough to test
- type changes in graphql-request
- some updates to jest and the testing framework

The `graphql-request` update is constrained to `5.1.0` because of an issue as noted by these issues:
- [467 in graphql-request](https://github.com/jasonkuhrt/graphql-request/issues/467)
- [9046 in graphql-code-generator](https://github.com/dotansimha/graphql-code-generator/issues/9046)

# 🧪 How has/should this change been tested? 

Check:
- client runs and the app works as expected
- check date pickers
- datatables are scrolling using the viewport
- the typescript validation is working (`yarn pre-commit-lint`)
- codgen works (`yarn generate`)
- storybook is working (`yarn storybook`)
- the translation & localisation is working
- android builds and runs

## 📃 Documentation

_No user facing changes_

### CODE REVIEW

File: client/packages/common/src/utils/types/TypeUtils.ts
Comment (CarlosNZ) [inline client/packages/common/src/utils/types/TypeUtils.ts:2]: Not strictly related, but should this method also check that the `variableToCheck` *is* an object before the `in` check? Then it doesn't need the `extends` as `T` could be any type.
```
typeof variableToCheck === 'object' &&
    variableToCheck !== null &&
    field in variableToCheck,
```

Also, this method doesn't actually match the type, it just checks that the property exists. The property itself could still be of a different type to the expected value, so this is somewhat misleadingly named. (`hasProperty` perhaps?). I'm actually not sure what this method does that is any different to just doing `<field> in variable` in place. (It's only used once as far as I can tell, and in that case using `if "expiryDate" in rowData` seems to do exactly the same thing and is arguably more straightforward). 

Do we need it? 

Sorry, bit of a tangent -- feel free to ignore 😆

**Review by CarlosNZ — COMMENTED** [review]
Nice, thanks. See comments below: 
- [X]     client runs and the app works as expected
- [ ]     check date pickers -- entering dates manually seems broken now. However, I've done some changes in Programs branch to improve this anyway, so I can replicate that here if you like. Although, I'll need to pull this into feature branch to check how this works with my changes, so we'll need to co-ordinate this somehow. Is it just the Date pickers package? I could update just that on my branch and see how it goes?
- [ ]     datatables are scrolling using the viewport -- something weird going on that I just noticed, although it's already like this in develop too. When scrolling down, the alternating row background colour is swapping back-and-forth, like the background is staying in place and it's applying it to whatever row is there rather than keeping the colour with the row. Presume this is something to do with the viewport filter, since each row it doesn't know what number row it is *overall*, perhaps? Like this:
Screenshot 2023-03-17 at 11 46 49 AM
Screenshot 2023-03-17 at 11 46 55 AM

- [x]     the typescript validation is working (yarn pre-commit-lint)
- [X]     codgen works (yarn generate)
- [ ]     storybook is working (yarn storybook) -- I'm getting a "Broken build" error here:
```
Error: [BABEL]: Cannot find module 'babel-plugin-dynamic-import-node/utils'
```

- [X]     the translation & localisation is working
- [ ]     android builds and runs (haven't tested)

**Comment by mark-prins** [comment]
>  I'm getting a "Broken build" error here:

Not getting that. I've just switched back to this branch and run `yarn install` then `yarn storybook` and it's all good. 
Perhaps `rm -rf node_modules`?

File: client/packages/common/src/utils/types/TypeUtils.ts
Comment (mark-prins) [reply client/packages/common/src/utils/types/TypeUtils.ts:2]: ideally that `isTypeOf` would be checking both the existence of the property and that it is the correct type. But since it doesn't.. I've removed. I think we have the same functionality with less code and more clarity with the change. Thanks!

**Comment by mark-prins** [comment]
> check date pickers

I thought that I'd tested manual entry 🤷  behaving strangely now. Yes, it's just the `@mui/x-date-pickers` which have been bumped up to the latest. If you have a fix that'd be great!

>  datatables are scrolling using the viewport 

Yeah, Chris noticed this too. I decided to ignore it. the visible rows are rendered new each time you scroll, so it's possible that the highlight colours would change.. unless perhaps we colour the rows not using css? that would be a new issue though, I wouldn't want to tackle that as part of this one.

File: client/packages/common/src/intl/strings/IntlStrings.ts
Comment (andreievg) [inline client/packages/common/src/intl/strings/IntlStrings.ts:6]: export TFunction and UseTranslationResponse seem to be unused, I thought it would have a small effect on bundle size increase but doesn't.

Looks like 10% increase is due to: https://github.com/jasonkuhrt/graphql-request/issues/446 (I looked at bundle analyser in 'static' view and found that graphql is much bigger, quick search of bundle size in their repo revealed the above).

I guess this is ok, as long as we are aware of it

Comment (mark-prins) [reply client/packages/common/src/intl/strings/IntlStrings.ts:6]: oops. have removed the `TFunction` that was a remnant from an attempted rewrite

**Review by andreievg — APPROVED** [review]
Looks good to me, I'll do the tests on the last PR

As per earlier discussion, will we merge this to develop, after Today/Tomorrow release ?

File: client/packages/common/src/intl/strings/IntlStrings.ts
Comment (mark-prins) [reply client/packages/common/src/intl/strings/IntlStrings.ts:6]: and bundle size.. yeah. and darn.
I specifically didn't update `graphql-request` because of that bug, as noted in the description. we'd have to sacrifice codegen in order to get the bundle size down. Or at least manually update all of the generated files.

I've tried the manual approach and the bundle is 2.24MB what do you think? shall we stick with it? Just means that running `yarn generate` will break things until we can get graphql-request updated

Comment (mark-prins) [reply client/packages/common/src/intl/strings/IntlStrings.ts]: put the change in 1c2d1ad81 😉

**Comment by mark-prins** [comment]
> As per earlier discussion, will we merge this to develop, after Today/Tomorrow release ?

Yes, good point. will hold off until we have a stable release

**Comment by CarlosNZ** [comment]
> I thought that I'd tested manual entry 🤷 behaving strangely now. Yes, it's just the `@mui/x-date-pickers` which have been bumped up to the latest. If you have a fix that'd be great!

I'll update the x-date-pickers in my branch and see what effect that is and report back.

**Comment by CarlosNZ** [comment]
> > I thought that I'd tested manual entry 🤷 behaving strangely now. Yes, it's just the `@mui/x-date-pickers` which have been bumped up to the latest. If you have a fix that'd be great!
> 
> I'll update the x-date-pickers in my branch and see what effect that is and report back.

Okay, it looks like it will work with my changes, so how about I replicate the relevant tweaks in this branch and make a new PR for it?

**Comment by CarlosNZ** [comment]
> > > I thought that I'd tested manual entry 🤷 behaving strangely now. Yes, it's just the `@mui/x-date-pickers` which have been bumped up to the latest. If you have a fix that'd be great!
> > 
> > 
> > I'll update the x-date-pickers in my branch and see what effect that is and report back.
> 
> Okay, it looks like it will work with my changes, so how about I replicate the relevant tweaks in this branch and make a new PR for it?

Okay, here you go: https://github.com/openmsupply/open-msupply/pull/1394

**Pull request merged**
